app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -12,35 +12,108 @@ import androidx.compose.material3.Scaffold
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.CompositionLocalProvider
 import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.tooling.preview.Preview
 import com.wire.android.R
 import com.wire.android.ui.common.bottomsheet.MenuBottomSheetItem
 import com.wire.android.ui.common.bottomsheet.MenuItemIcon
 import com.wire.android.ui.common.bottomsheet.MenuModalSheetLayout
 import com.wire.android.ui.common.snackbar.SwipeDismissSnackbarHost
+import com.wire.android.ui.common.WireDialog
+import com.wire.android.ui.common.WireDialogButtonProperties
+import com.wire.android.ui.common.WireDialogButtonType
+import com.wire.android.ui.common.button.WireButtonState
 import com.wire.android.ui.home.conversations.mock.mockMessages
 import com.wire.android.ui.home.conversations.model.AttachmentBundle
 import com.wire.android.ui.home.conversations.model.Message
 import com.wire.android.ui.home.conversations.model.MessageSource
 import com.wire.android.ui.home.messagecomposer.MessageComposer
 import kotlinx.coroutines.launch
+import com.wire.android.util.dialogErrorStrings
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun ConversationScreen(
     conversationViewModel: ConversationViewModel
 ) {
     val uiState = conversationViewModel.conversationViewState
-
+    val deleteMessageDialogState = conversationViewModel.deleteMessageDialogsState
     ConversationScreen(
         conversationViewState = uiState,
         onMessageChanged = { message -> conversationViewModel.onMessageChanged(message) },
         onSendButtonClicked = { conversationViewModel.sendMessage() },
         onSendAttachment = { attachmentBundle -> conversationViewModel.sendAttachmentMessage(attachmentBundle) },
-        onBackButtonClick = { conversationViewModel.navigateBack() }
+        onBackButtonClick = { conversationViewModel.navigateBack() },
+        onDeleteMessage = conversationViewModel::showDeleteMessageDialog
     )
+    if (deleteMessageDialogState is DeleteMessageState.State) {
+        when {
+            deleteMessageDialogState.deleteMessageDialogState is DeleteMessageDialogState.Visible -> {
```

Comment:
I think we can move this check one place which would be

DeleteMessageDialog()

Reply:
Done!